### ISSUE TITLE
.CountAsync() and .FirstAsync() throws exception when using ToInjectable

### ISSUE THREAD

**Issue by Shaddix** [axelheer/nein-linq#18]
The following code
```
var users = await _dbContext.Users
                .ToInjectable()
                .Select(x => new UserDto())
                .CountAsync();
```

throws
```
System.InvalidOperationException: The provider for the source IQueryable doesn't implement IAsyncQueryProvider. Only providers that implement IAsyncQueryProvider can be used for Entity Framework asynchr
onous operations.
   at Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions.ExecuteAsync[TSource,TResult](MethodInfo operatorMethodInfo, IQueryable`1 source, Expression expression, CancellationToken cancella
tionToken)
   at Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions.ExecuteAsync[TSource,TResult](MethodInfo operatorMethodInfo, IQueryable`1 source, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions.CountAsync[TSource](IQueryable`1 source, CancellationToken cancellationToken)

```
The same exception is for `.FirstAsync()` and `.FirstOrDefaultAsync()`.
On the contrary, `.Count()`, `.ToList()` or even `.ToListAsync()` work ok.

I'm using NeinLinq.EntityFrameworkCore version 4.1.0.

**Comment by peterwurzinger** [comment]
Using the EFCore specific flavor of NeinLinq should fix that:
```
var users = await _dbContext.Users
                .ToEntityInjectable()
                .Select(x => new UserDto())
                .CountAsync();
```

``ToEntityInjectable()`` lives in NeinLinq.EntityFrameworkCore and basically satifies EFCores "requirements" (*hrumph, hrumph*) regarding (asynchronous) query execution.
@axelheer correct me if I'm wrong.

**Comment by axelheer** [comment]
That's right, @peterwurzinger 👍

**Comment by Shaddix** [comment]
Thanks a lot, `ToEntityInjectable()` works!
I'm guessing, that EF is the most popular use-case for `nein-linq`, if it's really so, maybe it makes sense to extend documentation/Readme? 
Something like:

***Breaking:*** since Version `3.0.0` the extension methods have different names depending on the chosen package above, in order to avoid some conflicts! For example there are now `ToInjectable` (plain linq), `ToAsyncInjectable` (interactive), `ToDbInjectable` (EF6), and `ToEntityInjectable` (EFCore).
Usage of specific providers is encouraged for EF6/EFCore (otherwise async queries won't work).



Personally, I only added `NeinLinq.EntityFrameworkCore` nuget package, so when I used `.ToInjectable()` and it was successfully found, I thought I'm using extension method designed for EFCore.
